### ISSUE TITLE
The image format "Thumb" should overrule the Dimensions parameter

### ISSUE THREAD

**Issue by jongfeli** [yaronkoren/TinyMCE#5]
## On MW-1.27.4, MW 1.29.2 & MW 1.30.0
When selecting the image Format **Thumb** the Dimensions parameter needs to be overruled. 

![image](https://user-images.githubusercontent.com/6680027/38860574-c8393008-4230-11e8-9f7c-d89e5f040dae.png)

The result is:
`[[File:20171109 172708.jpg|300px|none|middle|thumb|20171109 172708.jpg]]`

It should be:
`[[File:20171109 172708.jpg|none|middle|thumb|20171109 172708.jpg]]`

No dimensions should be added. At this moment the Dimensions parameter is overruling the Thumb parameter, which in my opinion is not correct.

When a user changes the Thumb size in their preferences (after adding pictures to a page) the picture size will not change because the original thumb size will be used. Also other users will not see there thumb size but the size of the user that added the pictures to the page.

In short, Thumb does not need a size it is user specific.

**Comment by duncan-crane** [comment]
Hi Felipe

I think I understand you better now.  To recap my understanding:

Unless the user explicitly over-rides the size using the dimensions form field, you wish to use the user's default thumbsize for the html display in TinyMCE and just to use the Thumb parameter to control the size of the file displayed by the wiki code?  Both the latter should be the same for any given user, but could vary between different users, depending on the default thumbsize they have selected.

The reason I chose the approach I did was because it provided a fixed layout for the page, regardless of who was using it, which is perhaps what I am more used to when using word processors.  As you and Yaron are much more familiar with how people use wikis, I'm happy to take your concensus view as to the best approach?

Duncan

**Comment by jongfeli** [comment]
Hello Duncan

I am only a mere mortal but to me the `Thumb` parameter is there so the users can select there own preferred picture size. You can add sizes in `LocalSettings.php ` with the following example:

```php
## Adjust thumbnails on image pages according to a user setting. In order to
## reduce disk usage, the values can only be selected from a list. This is the
## list of settings the user can choose from:
$wgThumbLimits = array(
	120,
	150,
	180,
	200,
	250,
	300,
	350,
	400,
	500,
	600,
	700,
	800,
	900,
	1000
);
```  

The way it works now is that the thumbnail size of the user that adds the picture to the page dictates the size of the picture. If that user has a default size of 700px but other users are using 350px they still see a 700px picture.

But we could have both by respecting the `Dimensions` parameter at all time. If the `Dimensions` parameter is empty we should not assign any size to the picture no matter what. You then leave it up to the user if he uses `Thumb` or `Dimensions ` or both for that matter.

**Comment by duncan-crane** [comment]
Hi Felipe

I've made quite a lot of changes to this functionality over the weekend (see issue #3). This approach means that whatever formatting that mediawiki applies to the image should be the same as displayed in TinyMCE unless I've overlooked any css definitions.

Please do let me know if you spot any differences.

Many thanks
Duncan

**Comment by jongfeli** [comment]
What can I say... :+1: that means it works like a charm :grin:
